packages/react-native-editor/bin/extract-used-strings.js

```
@@ -0,0 +1,179 @@
+		const items = getStringsFromContext( po.translations[ context ] );
+		return [ ...result, ...items ];
+	}, [] );
+};
+
+const createString = ( previousString, string, platform ) =>
+	previousString
+		? {
+				...previousString,
+				platforms: [ ...previousString.platforms, platform ],
+		  }
+		: {
+				...string,
+				platforms: [ platform ],
+		  };
+
+const reduceStringsByPlatform = ( platform ) => ( result, item ) => ( {
+	...result,
+	[ item.string ]: createString( result[ item.string ], item, platform ),
+} );
+
+const getUsedStringsFromDomain = ( potFilesDir, domain ) => {
+	const androidUsedStrings = getStringsFromPotFile(
+		path.join( potFilesDir, `${ domain }-used-android.pot` )
+	);
+	const iosUsedStrings = getStringsFromPotFile(
+		path.join( potFilesDir, `${ domain }-used-ios.pot` )
+	);
+
+	let usedStrings = {};
+	usedStrings = androidUsedStrings.reduce(
+		reduceStringsByPlatform( 'android' ),
+		usedStrings
+	);
+	usedStrings = iosUsedStrings.reduce(
+		reduceStringsByPlatform( 'ios' ),
+		usedStrings
+	);
+
+	const sourceStrings = getStringsFromPotFile(
+		path.join( potFilesDir, `${ domain }-source.pot` )
+	);
+	const nativeRefRegex = /(native|ios|android)\.js(\:\d+)?$/;
+	usedStrings = sourceStrings
+		.filter(
+			( { string, reference } ) =>
+				!! usedStrings[ string ] &&
+				! reference.every( ( ref ) => nativeRefRegex.test( ref ) )
+		)
+		.reduce( reduceStringsByPlatform( 'web' ), usedStrings );
+
+	return usedStrings;
+};
+
+const getUsedStrings = ( potFilesDir, domains ) => {
+	return domains.reduce(
+		( result, domain ) => ( {
+			...result,
+			[ domain ]: getUsedStringsFromDomain( potFilesDir, domain ),
+		} ),
+		{}
+	);
+};
+
+const generatePotFiles = ( plugins, potFilesDir ) => {
+	const potFilesCommand = path.join( __dirname, 'generate-pot-files.sh' );
+	const potFilesArgs = plugins
+		.map( ( { name, sourcePath } ) => `${ name } ${ sourcePath }` )
+		.join( ' ' );
+	try {
+		// TODO: Detect potential parse failures
```

Ideally, we should detect from the output of the `generate-pot-files` script potential parse failures. Currently, these failures are only logged when using the debug mode so they're silently skipped.